src/EventListener/CreateSchemaListener.php

```diff
@@ -143,12 +145,29 @@ public function postGenerateSchema(GenerateSchemaEventArgs $eventArgs): void
 
     private function createForeignKeys(Table $relatedTable, Table $revisionsTable): void
     {
-        $revisionForeignKeyName = $this->config->getRevisionFieldName().'_'.md5($relatedTable->getName()).'_fk';
+        $revisionForeignKeyName = $this->config->getRevisionFieldName().'_'.\md5($relatedTable->getName()).'_fk';
         $primaryKey = $revisionsTable->getPrimaryKey();
         \assert(null !== $primaryKey);
 
+        /**
+         * doctrine/dbal 3 support - Table::addForeignKeyConstraint() takes a Table instead of a string
+         *
+         * NEXT_MAJOR: remove this `if` block
+         */
+        if (\version_compare(InstalledVersions::getVersion('doctrine/dbal') ?? '', '4.0.0', '<')) {
```

Comment:
Let's try to avoid using `InstalledVersions::getVersion` ; I don't think we ever used it on sonata/project.

What can be done, is something as `class_exists(Foo::class)` (or `!class_exists(Bar::class)` when Foo was added in 4.0 or Bar removed in 4.0.